### ISSUE TITLE
Safari v12 and v13 error for package versions 3.0.0 and above

### ISSUE THREAD

**Issue by tylerexmg** [launchdarkly/js-client-sdk#267]
My project updated from npm package v2 to v3 and it is causing a breaking error in Safari 12 and 13.

<img width="362" alt="image" src="https://user-images.githubusercontent.com/35921381/217589684-9b8a3e2b-faff-48b8-86a9-c32a46c96f8c.png">

Unfortunately I'm having a hard time debugging beyond this error to see exactly what the problem is. I just can confirm that using "launchdarkly-js-client-sdk": "^3.0.0" errors but "launchdarkly-js-client-sdk": "^2.24.2" does not.

**Comment by yusinto** [comment]
The LaunchDarkly JS SDK v3.0 and up uses optional chaining which is supported by modern browsers. It looks like you bundle your react app using webpack. The issue is webpack 4 uses a version of [acorn](https://github.com/webpack/webpack/issues/10227#issuecomment-706752571) which does not transpile the optional chaining operator. The solution as you can see in the react async provider [example](https://github.com/launchdarkly/react-client-sdk/blob/c840be20ab8d370864eab4c6bb2062918efaa891/examples/async-provider/package.json#L63) is to resolve the acorn package version that does support it. To do this, in your package.json file:

```json
  "resolutions": {
    "acorn": "npm:acorn-with-stage3"
  }
```

Webpack 5 supports optional chaining out of the box so we recommend you upgrade to that if possible. If you use other bundlers, make sure to enable/include plugins for optional chaining for your bundler.

This is also covered in our v2 to v3 migration docs. Read more [here](https://docs.launchdarkly.com/sdk/client-side/javascript/migration-2-to-3?q=js%20sdk#dependencies-for-the-v3-sdk).

**Comment by tylerexmg** [comment]
Hi @yusinto thank you for your reply. We are using Webpack 5 `"webpack": "^5.73.0"` and also use optional chaining in our code without issue when reverting to the LD JS SDK v2.

**Comment by yusinto** [comment]
This should be fixed in [3.1.1](https://github.com/launchdarkly/js-client-sdk/releases/tag/3.1.1). We have removed the optional chaining operator so please update to that version if you have issues with the `?` operator.
